popup.js

```diff
@@ -0,0 +1,167 @@
+          <div>
+              <h3 class='card-detail-heading'>${project.name}</h3>
+              <div class='d-flex works-card-client'>
+                  <p class='paragraph'>${project.details.company}</p>
+                  <i class='fa-solid fa-circle my-works-card-client-counter'></i>
+                  <p class='works-card-client-role paragraph'>${
+  project.details.profile
+}</p>
+                  <i class='fa-solid fa-circle my-works-card-client-counter'></i>
+                  <p class='my-works-card-client-year paragraph'>${
+  project.details.year
+}</p>
+              </div>
+          </div>
+          <i id='card-detail-close-${
+  index + 1
+}' onclick="hideProjectDetails(${index + 1})" class='fa-solid fa-xmark card-detail-cross-icon'></i>
+      </div>
+      <div class='card-detail-display-img-container'>
+          <img class='' src='${project.featuredImg}' alt='Recent Work'>
+      </div>
+      <div class='card-detail-body'>
+          <p class='paragraph'>${project.Description}</p>
+          <div class='card-detail-body-right'>
+              <ul class="d-flex card-detail-body-tags">
+                <li class="card-detail-body-tag">${project.language[0]}</li>
+                <li class="card-detail-body-tag">${project.language[1]}</li>
+                <li class="card-detail-body-tag">${project.language[2]}</li>
+              </ul>
+              <div class='card-detail-body-right-button-container'>
+                  <button class='card-detail-button' src='${
+  project.seeLive
+}'>See Live
+                   <i class='fa-solid fa-arrow-up-right-from-square'></i></button>
+                  <button class='card-detail-button'  src='${
+  project.seeSource
+}'>
+                  See Source <i class='fa-brands fa-github'></i></button>
+              </div>
+          </div>
+      </div>
+    </div>`;
+  return popup;
+}
+
+function showProjectDetails(index) {
+  const popup = document.getElementById(`card-detail-${index}`);
+  const body = document.getElementById('body');
+  popup.classList.add('card-detail--show');
+  body.classList.add('blur');
+}
+
+function hideProjectDetails(index) {
+  const popup = document.getElementById(`card-detail-${index}`);
+  const body = document.getElementById('body');
+  popup.classList.remove('card-detail--show');
+  body.classList.remove('blur');
+}
+
+function renderProjects() {
+  const container = document.getElementById('works-cards');
+  const popupContainer = document.getElementById('popup-container');
+
+  projects.forEach((project, index) => {
+    const card = createCard(project, index);
+    container.appendChild(card);
+
+    const popup = createPopup(project, index);
+    popupContainer.appendChild(popup);
+
+    card.addEventListener('click', () => {
+      showProjectDetails(index);
+    });
+
+    popup.querySelector(`#card-detail-close-${index}`).addEventListener('click', () => {
+      hideProjectDetails(index);
+    });
+  });
+}
```

Comment:
I noticed you have 4 projects in your projects array, but the work card and popup card are only being rendered for the first object in your array, only one project is showing:

![127 0 0 1_5502_](https://github.com/the-faizmohammad/my-portfolio/assets/93520178/8a013a46-9971-46ec-8e37-3183b972e325)

Kindly fix this issue.